MBS-11527 Bump up hamcrest version

RuslanMingaliev opened the pull request.

RuslanMingaliev committed:
MBS-11527 Bump up hamcrest version

RuslanMingaliev committed:
MBS-11527 Bump up hamcrest version

RuslanMingaliev committed:
MBS-11631 Use bundle for hamcrest

RuslanMingaliev committed:
MBS-11631 Use bundle for hamcrest

eugene-krivobokov commented on gradle/libs.versions.toml:
```diff
@@ -100,3 +102,6 @@ jdgraphtCore = "org.jgrapht:jgrapht-core:1.5.1"
 detektFormatting = { module = "io.gitlab.arturbosch.detekt:detekt-formatting", version.ref = "detekt" }
+
+[bundles]
+hamcrest = ["hamcrest", "hamcrestLibrary", "hamcrestCore"]
```
`hamcrest-library` depends on `hamcrest-core`: https://repo1.maven.org/maven2/org/hamcrest/hamcrest-library/2.2/hamcrest-library-2.2.pom
Why do we add transitive dependency explicitly to clients?

RuslanMingaliev replied:
There is no reason to do that, thanks

RuslanMingaliev committed:
MBS-11631 Merge branch 'develop' into MBS-11631-bump-hamcrest-version

RuslanMingaliev committed:
MBS-11631 Remove hamcrest-core from version catalog

RuslanMingaliev committed:
MBS-11631 Remove hamcrest-core from version catalog

RuslanMingaliev committed:
MBS-11631 Fix type inference

RuslanMingaliev committed:
MBS-11631 Fix type inference

eugene-krivobokov commented on gradle/libs.versions.toml:
```diff
@@ -78,7 +78,8 @@ dexlib = "org.smali:dexlib2:2.3"
 truth = "com.google.truth:truth:1.0"
-hamcrestLib = "org.hamcrest:hamcrest-library:1.3"
+hamcrest = "org.hamcrest:hamcrest:2.2"
+hamcrestLibrary = "org.hamcrest:hamcrest-library:2.2"
```
Should we extract hamcrest version to versions?

RuslanMingaliev committed:
MBS-11631 Use version.ref for hamcrest versions

RuslanMingaliev committed:
MBS-11631 Use version.ref for hamcrest versions

RuslanMingaliev committed:
Merge branch 'develop' into MBS-11631-bump-hamcrest-version

RuslanMingaliev committed:
Merge branch 'develop' into MBS-11631-bump-hamcrest-version

RuslanMingaliev committed:
MBS-11631 Merge remote-tracking branch 'origin/MBS-11631-bump-hamcrest-version' into MBS-11631-bump-hamcrest-version

# Conflicts:
#	gradle/libs.versions.toml
#	subprojects/common/junit-utils/build.gradle.kts

RuslanMingaliev committed:
MBS-11631 Fix merge issues
